python/pyspark/ml/classification.py

```diff
@@ -1117,6 +1137,56 @@ def getBlockSize(self):
         """
         return self.getOrDefault(self.blockSize)
 
+    @since("2.0.0")
+    def setStepSize(self, value):
+        """
+        Sets the value of :py:attr:`stepSize`.
+        """
+        return self._set(stepSize=value)
+
+    @since("2.0.0")
+    def getStepSize(self):
+        """
+        Gets the value of stepSize or its default value.
+        """
+        return self.getOrDefault(self.stepSize)
+
+    @since("2.0.0")
+    def setSolver(self, value):
+        """
+        Sets the value of :py:attr:`solver`.
+        """
+        return self._set(solver=value)
+
+    @since("2.0.0")
+    def getSolver(self):
+        """
+        Gets the value of solver or its default value.
+        """
+        return self.getOrDefault(self.solver)
+
+    @property
+    @since("2.0.0")
+    def getOptimizer(self):
```

Comment:
Sorry want clear - yeah agree it should match Scala. But maybe we should
just deprecate it, seems superfluous.

On Thu, 12 May 2016 at 00:11, Holden Karau <notifications@github.com> wrote:

> In python/pyspark/ml/classification.py
> <https://github.com/apache/spark/pull/12943#discussion_r62935883>:
>
> > +    def setSolver(self, value):
> > +        """
> > +        Sets the value of :py:attr:`solver`.
> > +        """
> > +        return self._set(solver=value)
> > +
> > +    @since("2.0.0")
> > +    def getSolver(self):
> > +        """
> > +        Gets the value of solver or its default value.
> > +        """
> > +        return self.getOrDefault(self.solver)
> > +
> > +    @property
> > +    @since("2.0.0")
> > +    def getOptimizer(self):
>
> It is somewhat of a duplicate but I just added it to match the Scala API.
> I'm pretty ambivalent on the actual method its self.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12943/files/7cc27e91a9cfd234268c0b3bb3e05c22dddb8ce0#r62935883>
>